www/js/index.js

```diff
@@ -1,49 +1,225 @@
- * regarding copyright ownership.  The ASF licenses this file
- * to you under the Apache License, Version 2.0 (the
- * "License"); you may not use this file except in compliance
- * with the License.  You may obtain a copy of the License at
- *
- * http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing,
- * software distributed under the License is distributed on an
- * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
- * KIND, either express or implied.  See the License for the
- * specific language governing permissions and limitations
- * under the License.
- */
-var app = {
-    // Application Constructor
-    initialize: function() {
-        this.bindEvents();
-    },
-    // Bind Event Listeners
-    //
-    // Bind any events that are required on startup. Common events are:
-    // 'load', 'deviceready', 'offline', and 'online'.
-    bindEvents: function() {
-        document.addEventListener('deviceready', this.onDeviceReady, false);
-    },
-    // deviceready Event Handler
-    //
-    // The scope of 'this' is the event. In order to call the 'receivedEvent'
-    // function, we must explicitly call 'app.receivedEvent(...);'
-    onDeviceReady: function() {
-        app.receivedEvent('deviceready');
-    },
-    // Update DOM on a Received Event
-    receivedEvent: function(id) {
-        var parentElement = document.getElementById(id);
-        var listeningElement = parentElement.querySelector('.listening');
-        var receivedElement = parentElement.querySelector('.received');
-
-        listeningElement.setAttribute('style', 'display:none;');
-        receivedElement.setAttribute('style', 'display:block;');
-
-        console.log('Received Event: ' + id);
-    }
-};
+$(function($) {
+
+    var ROUTES_BY_STOP_NAME_ENDPOINT = 'https://api.appglu.com/v1/queries/findRoutesByStopName/run',
+        STOPS_BY_ROUTE_ID_ENDPOINT = 'https://api.appglu.com/v1/queries/findStopsByRouteId/run',
+        DEPARTURES_BY_ROUTE_ID_ENDPOINT = 'https://api.appglu.com/v1/queries/findDeparturesByRouteId/run',
+        USERNAME = 'WKD4N7YMA1uiM8V',
+        PASSWORD = 'DtdTtzMLQlA0hk2C1Yi5pLyVIlAQ68',
+        DEFAULT_AJAX_OPTIONS = {
+            method: 'POST',
+            dataType: 'json',
+            contentType: 'application/json',
+            headers: {
+                'X-AppGlu-Environment': 'staging',
+                // about JS's `btoa` function:
+                // - https://developer.mozilla.org/en-US/docs/Web/API/WindowBase64/btoa
+                // - http://caniuse.com/#feat=atob-btoa
+                'Authorization': "Basic " + btoa(USERNAME + ":" + PASSWORD)
+            },
+        },
+        emptyFunc = function() {};
+
+    function searchStopsByRouteId(routeId, args) {
+
+        args = $.extend({}, {
+            beforeStart: emptyFunc
+        }, args);
+        var beforeStart = args.beforeStart;
+
+        var options = $.extend({}, DEFAULT_AJAX_OPTIONS, {
```

Comment:
Those functions seem a bit of repetitive code to me. How could you remove this duplication of code?